File: fru1tworld/prev/week21~25/week24/leet746/leet746.cpp
class Solution {
    public:
        int minCostClimbingStairs(vector<int>& cost) {
            int len = cost.size();
            if(len == 1) return 0; 
            vector<int> dp(len);
            dp[0] = cost[0];
            dp[1] = cost[1];
            for(int i=2; i<len; i++){
                dp[i] = cost[i] + min(dp[i-1], dp[i-2]);
            }
            return min(dp[len-1], dp[len-2]);
        }
    };

File: iam454/week24/leet746.js
/**
 * @param {number[]} cost
 * @return {number}
 */
var minCostClimbingStairs = function (cost) {
  const N = cost.length;
  const dp = new Array(N).fill(0);

  dp[0] = cost[0];
  dp[1] = cost[1];

  for (let i = 2; i < N; i++) {
    dp[i] = Math.min(dp[i - 2] + cost[i], dp[i - 1] + cost[i]);
  }

  return Math.min(dp[N - 1], dp[N - 2]);
};
